operator/pkg/controller/istiocontrolplane/istiocontrolplane_controller.go

```diff
@@ -232,9 +232,8 @@ func (r *ReconcileIstioOperator) Reconcile(_ context.Context, request reconcile.
 	if iop.Spec == nil {
 		iop.Spec = &v1alpha1.IstioOperatorSpec{Profile: name.DefaultProfileName}
 	}
-	operatorRevision, _ := os.LookupEnv("REVISION")
-	if operatorRevision != "" && operatorRevision != iop.Spec.Revision {
-		scope.Infof("Ignoring IstioOperator CR %s with revision %s, since operator revision is %s.", iopName, iop.Spec.Revision, operatorRevision)
+	if operatorRevision, revisionEnvExists := os.LookupEnv("REVISION"); revisionEnvExists && operatorRevision != iop.Spec.Revision {
```

Comment:
At the very least, I think REVISION="" should keep the capability to reconcile iop.Spec.Revision="default", otherwise it will be inconsistent with the existing installation revision semantics.

Reply:
Thanks @hanxiaop for reviewing. That makes sense. I have added the change. PTAL

Unrelated to this PR, I think there is one more place I recently came across this inconsistency https://github.com/istio/istio/blob/master/operator/pkg/helmreconciler/reconciler.go#L605